### DISCUSSION TITLE
Isolated event for spells on use/onhit

### DISCUSSION THREAD

**Discussion by PyroTech03** [AscensionGameDev/Intersect-Engine#2202] [Ideas]
Similar vein as #2181 but for spells.

Implement an isolated event similar to task completion on quests. This would allow some event implementation without taking up the spells capability with the "Event" type.

You could use this to make spells that level up or implement a skill system for magic.

**Comment by cydyn** [comment]
@PyroTech03 I can possibly do it, because I'm testing it, so that common events have new triggers for passive spells. onhitmelee, onhitall and recivedamage. Then we only set the appropriate trigger in the common event and use the requirements, e.g.: he must have learned the fire shield spell, and then the given common event is fired. onhitmelee - if we deal damage from an auto attack onhitall - if we deal any damage, regardless of whether it is a spell or an autoattack and recivedamage - if someone insults us.

**Reply by PyroTech03** [reply]
What day did for #2181 makes a good template for implementing this.

**Reply by cydyn** [reply]
I don't know the database well and to do what I want I would have to dig around there
